genienlp/paraphrase/transformers_utils.py

```diff
@@ -106,7 +101,28 @@ def set_tgt_lang_special_tokens(self, lang: str) -> None:
         self.cur_lang_code = self.lang_code_to_id[lang]
         self.prefix_tokens = [self.cur_lang_code]
         self.suffix_tokens = [self.eos_token_id]
-        
+
+
+###############
+
+
+class GenieMarianTokenizer(MarianTokenizer):
+    '''
+    MarianTokenizer with the temporary fix for decoding.
+    In current huggingface's implementation `convert_tokens_to_string` method always uses spm_target to decode.
+    To be able to decode both source language and target language, this class changes the spm to be current_spm
+    '''
```

Comment:
Have we filed an issue to get this upstream?

Reply:
I will do that

Reply:
Fixed upstream https://github.com/huggingface/transformers/pull/10357